ui/src/stores/utils/contract.js

```diff
@@ -62,70 +62,7 @@ export const totalBurntCoins = async contract => {
   return new BN(burntCoins)
 }
 
-export const getValidatorList = async (address, eth) => {
-  const validatorsContract = new eth.Contract(REWARDABLE_VALIDATORS_ABI, address)
-  const validators = await validatorList(validatorsContract)
-
-  if (validators.length) {
-    return validators
-  }
-
-  const deployedAtBlock = await getDeployedAtBlock(validatorsContract)
-  const contract = new eth.Contract([], address)
-  const validatorsEvents = await contract.getPastEvents('allEvents', {
-    fromBlock: Number(deployedAtBlock)
-  })
-
-  return processValidatorsEvents(validatorsEvents)
-}
-
-export const validatorList = async contract => {
-  try {
-    return await contract.methods.validatorList().call()
-  } catch (e) {
-    return []
-  }
-}
-
-export const processValidatorsEvents = events => {
-  const validatorList = new Set()
-  events.forEach(event => {
-    parseValidatorEvent(event)
-
-    if (event.event === 'ValidatorAdded') {
-      validatorList.add(event.returnValues.validator)
-    } else if (event.event === 'ValidatorRemoved') {
-      validatorList.delete(event.returnValues.validator)
-    }
-  })
-
-  return Array.from(validatorList)
-}
-
-export const parseValidatorEvent = event => {
-  if (
-    event.event === undefined &&
-    event.raw &&
-    event.raw.topics &&
-    (event.raw.topics[0] === '0xe366c1c0452ed8eec96861e9e54141ebff23c9ec89fe27b996b45f5ec3884987' ||
-      event.raw.topics[0] === '0x8064a302796c89446a96d63470b5b036212da26bd2debe5bec73e0170a9a5e83')
-  ) {
-    const rawAddress = event.raw.topics.length > 1 ? event.raw.topics[1] : event.raw.data
-    const address = '0x' + rawAddress.slice(26)
-    event.event = 'ValidatorAdded'
-    event.returnValues.validator = address
-  } else if (
-    event.event === undefined &&
-    event.raw &&
-    event.raw.topics &&
-    event.raw.topics[0] === '0xe1434e25d6611e0db941968fdc97811c982ac1602e951637d206f5fdda9dd8f1'
-  ) {
-    const rawAddress = event.raw.data === '0x' ? event.raw.topics[1] : event.raw.data
-    const address = '0x' + rawAddress.slice(26)
-    event.event = 'ValidatorRemoved'
-    event.returnValues.validator = address
-  }
-}
+export const getValidatorList = async (address, eth) => commonGetValidatorList(address, eth, { logger: console })
```

Comment:
The `getValidatorList` and `getPastEvents ` functions are left here (instead of removing it and importing from `commons` everywhere), because it is mocked here:

https://github.com/poanetwork/tokenbridge/blob/055a444faea09797bbe9b4dc1a29a2b514e9cd23/ui/src/stores/__tests__/HomeStore.test.js#L28-L36

And I failed to mock importing these from `commons` in `HomeStore` and `ForeignStore`